lib/membership/kubernetes/kubernetes_test.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetes
+
+import (
+	"sort"
+	"testing"
+
+	pb "github.com/gravitational/satellite/agent/proto/agentpb"
+	"github.com/gravitational/satellite/lib/test"
+
+	. "gopkg.in/check.v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/tools/cache"
+)
+
+func TestNodes(t *testing.T) { TestingT(t) }
+
+type KubernetesSuite struct{}
+
+var _ = Suite(&KubernetesSuite{})
+
+// TestMembers verfies members can be queried.
+func (r *KubernetesSuite) TestMembers(c *C) {
+	tests := []struct {
+		comment  string
+		nodes    []v1.Node
+		expected []*pb.MemberStatus
+	}{
+		{
+			comment: "List all nodes",
+			nodes: []v1.Node{
+				r.newNode("satellite-1", "192.168.1.101", "master"),
+				r.newNode("satellite-2", "192.168.1.102", "master"),
+			},
+			expected: []*pb.MemberStatus{
+				pb.NewMemberStatus("satellite-1", "192.168.1.101", map[string]string{"role": "master"}),
+				pb.NewMemberStatus("satellite-2", "192.168.1.102", map[string]string{"role": "master"}),
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		comment := Commentf(tc.comment)
+
+		factory := informers.NewSharedInformerFactory(fake.NewSimpleClientset(&v1.NodeList{Items: tc.nodes}), 0)
+		informer := factory.Core().V1().Nodes().Informer()
+
+		stop := make(chan struct{})
+		defer close(stop)
+		go informer.Run(stop)
```

Comment:
nit: this is going to get called only upon exit from the test method - have you considered confining the actual test in a separate function. In case more test cases are added, this might have a side-effect.
On the other hand, it looks like there's just a single test-case - why use table-driven testing prematurely?

Reply:
Good point. I guess there is no need to use table-driven testing here.